sys/net/gnrc/link_layer/lwmac/lwmac.c

```diff
@@ -0,0 +1,745 @@
+                LOG_INFO("Broadcast transmission done\n");
+            } else {
+                LOG_INFO("Transmission was %ssuccessful (%"PRIu32" WRs sent)\n",
+                         tx_success, lwmac.tx.wr_sent);
+            }
+            lwmac_tx_stop(&lwmac);
+            lwmac_set_state(SLEEPING);
+            break;
+        default:
+            lwmac_tx_update(&lwmac);
+        }
+
+        /* If state has changed, reschedule main state machine */
+        if(state_tx != lwmac.tx.state)
+        {
+            lwmac_schedule_update();
+        }
+        break;
+    }
+    default:
+        LOG_DEBUG("No actions in state %u\n", lwmac.state);
+    }
+
+    return lwmac.needs_rescheduling;
+}
+
+/******************************************************************************/
+
+static void rtt_cb(void* arg)
+{
+    msg_t msg;
+    msg.content.value = ((uint32_t) arg ) & 0xffff;
+    msg.type = LWMAC_EVENT_RTT_TYPE;
+    msg_send(&msg, lwmac.pid);
+
+    if (sched_context_switch_request) {
+        thread_yield();
+    }
+}
+
+/******************************************************************************/
+
+// TODO: Don't use global variables
+void rtt_handler(uint32_t event)
+{
+    uint32_t alarm;
+    switch(event & 0xffff)
+    {
+    case LWMAC_EVENT_RTT_WAKEUP_PENDING:
+        lwmac.last_wakeup = rtt_get_alarm();
+        alarm = _next_inphase_event(lwmac.last_wakeup, RTT_US_TO_TICKS(LWMAC_WAKEUP_DURATION_US));
+        rtt_set_alarm(alarm, rtt_cb, (void*) LWMAC_EVENT_RTT_SLEEP_PENDING);
+        lpm_prevent_sleep |= LWMAC_LPM_MASK;
+        lwmac_set_state(LISTENING);
+        break;
+
+    case LWMAC_EVENT_RTT_SLEEP_PENDING:
+        alarm = _next_inphase_event(lwmac.last_wakeup, RTT_US_TO_TICKS(LWMAC_WAKEUP_INTERVAL_US));
+        rtt_set_alarm(alarm, rtt_cb, (void*) LWMAC_EVENT_RTT_WAKEUP_PENDING);
+        lpm_prevent_sleep &= ~(LWMAC_LPM_MASK);
+        lwmac_set_state(SLEEPING);
+        break;
+
+    /* Set initial wakeup alarm that starts the cycle */
+    case LWMAC_EVENT_RTT_START:
+        LOG_DEBUG("RTT: Initialize duty cycling\n");
+        alarm = rtt_get_counter() + RTT_US_TO_TICKS(LWMAC_WAKEUP_DURATION_US);
+        rtt_set_alarm(alarm, rtt_cb, (void*) LWMAC_EVENT_RTT_SLEEP_PENDING);
+        lpm_prevent_sleep |= LWMAC_LPM_MASK;
+        lwmac.dutycycling_active = true;
+        break;
+
+    case LWMAC_EVENT_RTT_STOP:
+    case LWMAC_EVENT_RTT_PAUSE:
+        rtt_clear_alarm();
+        LOG_DEBUG("RTT: Stop duty cycling, now in state %u\n", lwmac.state);
+        lpm_prevent_sleep |= LWMAC_LPM_MASK;
+        lwmac.dutycycling_active = false;
+        break;
+
+    case LWMAC_EVENT_RTT_RESUME:
+        LOG_DEBUG("RTT: Resume duty cycling\n");
+        alarm = _next_inphase_event(lwmac.last_wakeup, RTT_US_TO_TICKS(LWMAC_WAKEUP_INTERVAL_US));
+        rtt_set_alarm(alarm, rtt_cb, (void*) LWMAC_EVENT_RTT_WAKEUP_PENDING);
+        lpm_prevent_sleep &= ~(LWMAC_LPM_MASK);
```

Comment:
Yes, roughly. `lpm_prevent_sleep` is a global wake lock. The system will go to into standby mode as soon as the idle thread gets scheduled the next time and no other peripheral or thread has aquired a wake lock.